### PR TITLE
Adding SAS token option to master template

### PR DESCRIPTION
**Pull request by MichalKosowski**
**What**
This pull request add option to Creator configuration yml file which can optionally include sas token parameter in the linked templates uri value.

**Why**
If we are using Azure DevOps, the default option to create linked templates is to linked them via Storage Account blob. But to make that working we need to include SAS token in the url (which can be easily obtained in release pipeline).
Currently this is not possible and master template needs to be edited manually or via some script (or other workaround techniques must be used)

**How**
When
`linked: true`
Then we can also specify:
`includeSasTokenVariable: true`

So that in master template linked templates uri will change from:
`"uri": "[concat(parameters('TemplatesStorageAccount'), 'Output/fleetmanagement-apim-dev-Client.api.template.json')]",`

to:
`"uri": "[concat(parameters('TemplatesStorageAccount'), 'Output/fleetmanagement-apim-dev-Client.api.template.json', parameters('templatesStorageAccountSASToken'))]",`

**Remarks**
I did not try to clean up the code which is generating master template, but it is not very clean. however I changed one bit, in which master template generating method can accept parameter from previously created mastertemplatesparameters file setup. So that there is no need to duplicate logic. However I kept old logic there. I done minimal changes which are not great, but refactoring can be done in separate pull request, eventually. This is only about a feature which is missing for me, really.

### CODE REVIEW

**Comment by MichalKosowski** [comment]
hmm looks like I have messed with some commits history (probably), sorry for that. I will be more careful next time. Or if that is a big issue, I can recreate a feature branch to make it more clean...

File: src/APIM_ARMTemplate/apimtemplate.test/Creator/TemplateCreatorTests/MasterTemplateCreatorTests.cs [hunk 1]
```diff
@@ -24,7 +24,7 @@ public void ShouldCreateCorrectNumberOfDeploymentResources()
             int count = 5;
 
             // act
-            Template masterTemplate = masterTemplateCreator.CreateLinkedMasterTemplate(apiVersionSetsTemplate, productsTemplate, loggersTemplate, null, null, apiInfoList, creatorFileNames, creatorConfig.apimServiceName, fileNameGenerator);
+            Template masterTemplate = masterTemplateCreator.CreateLinkedMasterTemplate(apiVersionSetsTemplate, productsTemplate, loggersTemplate, null, null, apiInfoList, creatorFileNames, creatorConfig.apimServiceName, fileNameGenerator, false, null);
```
Comment (anganti) [inline src/APIM_ARMTemplate/apimtemplate.test/Creator/TemplateCreatorTests/MasterTemplateCreatorTests.cs]: Lets try use named arguments. Improves readability

File: src/APIM_ARMTemplate/apimtemplate/apimtemplate.csproj [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (anganti) [inline src/APIM_ARMTemplate/apimtemplate/apimtemplate.csproj]: no change

File: src/APIM_ARMTemplate/apimtemplate.test/Creator/TemplateCreatorTests/MasterTemplateCreatorTests.cs (hunk 1 above)
Comment (MichalKosowski) [reply src/APIM_ARMTemplate/apimtemplate.test/Creator/TemplateCreatorTests/MasterTemplateCreatorTests.cs]: done. howerver this is a test class code so not so important. I think the whole method have too many parameters already and some clean up will be required sooner or later.

**Comment by lucashuet93** [comment]
Hi Michal, thanks so much for your contribution. We've internally come across the same need to accommodate private storage and have had discussions about the best way to fix this issue. While many users will use private blob storage containers that necessitate SAS token accommodation, they could also use a variety of other remote storage options, so our plan was to generalize this and support all query string options rather than specifically support SAS tokens. We've opened issues #235  and #236, which will be PR'd into master shortly. 

Thanks again for your work, but I'm closing this PR as a result.

**Pull request closed without merging**